src/TropicalGeometry/variety_prime.jl

```diff
@@ -0,0 +1,135 @@
+################################################################################
+#
+#  Tropicalization of prime ideals
+#
+#  WARNING: assumes without test that `I` is prime
+#
+################################################################################
+
+function tropical_variety_prime(I::MPolyIdeal, nu::TropicalSemiringMap; weighted_polyhedral_complex_only::Bool=false)
+    # compute a reduced GB to check whether I is homogeneous
+    G = groebner_basis(I,complete_reduction=true)
+
+    if all(Oscar._is_homogeneous.(G))
+        return tropical_variety_prime_singular(I,nu; weighted_polyhedral_complex_only=weighted_polyhedral_complex_only)
+    end
+
+    Ih = homogenize_pre_tropicalization(I)
+    TropIh = tropical_variety_prime_singular(Ih,nu,weighted_polyhedral_complex_only=weighted_polyhedral_complex_only)
+    return dehomogenize_post_tropicalization(TropIh)
+end
+
+# trivial valuation
+function tropical_variety_prime_singular(I::MPolyIdeal, nu::TropicalSemiringMap{QQField,Nothing,<:Union{typeof(min),typeof(max)}}; weighted_polyhedral_complex_only::Bool=false)
+    R = base_ring(I)
+    singularCommand = join(["ring r=0,("*join(string.(symbols(R)),",")*"),dp;",
+                            "ideal I = "*join(string.(gens(I)), ",")*";",
+                            "if (!defined(tropicalVariety)) { LIB \"tropical.lib\"; };",
+                            "fan TropI = tropicalVariety(I);",
+                            "string TropIString = string(TropI);"])
+    Singular.call_interpreter(singularCommand)
+    TropIString = Singular.lookup_library_symbol("Top", "TropIString")
+    Sigma = gfan_fan_string_to_oscar_complex(TropIString,convention(nu)==max,false)
+    TropI = compute_weights_and_construct_tropical_variety(Sigma,I,nu)
```

Comment:
Hmm, do you really need to go through the interpreter? can't you do something like this instead?
```suggestion
    I_sing = singular_generators(I)
    TropI = Singular.LibTropical.tropicalVariety(I_sing)
```
OK this probably is missing some steps, `TropI` maybe needs additional processing -- but I hope the gist is clear?

Reply:
Ah, sorry, I only now discovered you already tried this and submitted https://github.com/oscar-system/Singular.jl/issues/807

I'll comment there.